bin/blocks-plot

```diff
@@ -34,14 +34,14 @@ def plot_dataframe(dataframe):
     print("Plotting {} channels:".format(len(dataframe.columns)))
     for cname, series in iteritems(dataframe):
         print("    {}".format(cname))
-        pylab.plot(t, series.interpolate(), label=cname)
```

Comment:
Do you understand you right, that if I put a log record with training cost
every 10 iterations, the plot will be blank, because there are no two
successive vaild datapoints?

On 19 October 2015 at 09:29, Jörg Bornschein <notifications@github.com>
wrote:

> In bin/blocks-plot
> <https://github.com/mila-udem/blocks-extras/pull/28#discussion_r42370070>:
>
> > @@ -34,14 +34,14 @@ def plot_dataframe(dataframe):
> >      print("Plotting {} channels:".format(len(dataframe.columns)))
> >      for cname, series in iteritems(dataframe):
> >          print("    {}".format(cname))
> > -        pylab.plot(t, series.interpolate(), label=cname)
>
> There will be gaps in the lineplot where the NaNs are. Which is fine, as
> long as long as you have more than one successive valid datapoint (E.g.: an
> alternating Series with [1., Nan, 2., NaN, 3., NaN, ...] would be the worse
> case and not display anything)
>
> But using interpolate will extend all shorter graphs all the way to the
> right using a straight line.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/mila-udem/blocks-extras/pull/28/files#r42370070>.
>